src/type/__tests__/serialization-test.js

```diff
@@ -126,19 +117,41 @@ describe('Type System: Scalar coercion', () => {
     expect(GraphQLString.serialize(stringableObjValue)).to.equal(
       'something useful',
     );
+
+    expect(() => GraphQLString.serialize(NaN)).to.throw(
+      'String cannot represent value: NaN',
+    );
+
+    expect(() => GraphQLString.serialize([1])).to.throw(
+      'String cannot represent value: [1]',
+    );
+
+    const badObjValue = {};
+    expect(() => GraphQLString.serialize(badObjValue)).to.throw(
+      'String cannot represent value: {}',
+    );
   });
 
   it('serializes output as Boolean', () => {
-    expect(GraphQLBoolean.serialize('string')).to.equal(true);
-    expect(GraphQLBoolean.serialize('false')).to.equal(true);
-    expect(GraphQLBoolean.serialize('')).to.equal(false);
```

Comment:
Converting non-empty strings to `true` is confusing because `GraphQLString` coerce  `true` into `"true"`. Also spec doesn't mention this coercion:
http://facebook.github.io/graphql/June2018/#sec-Boolean